src/CbcCInterface.jl

```diff
@@ -136,32 +140,42 @@ function loadProblem(prob::CbcModel,
     mat = convert(SparseMatrixCSC{Float64,Int32},constraint_matrix)
     nrow,ncol = size(mat)
 
-    @cbc_ccall loadProblem Cvoid (Ptr{Cvoid}, Int32, Int32, Ptr{CoinBigIndex},
-        Ptr{Int32}, Ptr{Float64}, Ptr{Float64}, Ptr{Float64},
-        Ptr{Float64}, Ptr{Float64}, Ptr{Float64}) prob.p ncol nrow mat.colptr.-Int32(1) mat.rowval.-Int32(1) mat.nzval vec_or_null(Float64, col_lb, ncol) vec_or_null(Float64, col_ub, ncol) vec_or_null(Float64, obj, ncol) vec_or_null(Float64, row_lb, nrow) vec_or_null(Float64, row_ub, nrow)
+    GC.@preserve prob mat begin
+        @cbc_ccall loadProblem Cvoid (Ptr{Cvoid}, Int32, Int32, Ptr{CoinBigIndex},
+            Ptr{Int32}, Ptr{Float64}, Ptr{Float64}, Ptr{Float64},
+            Ptr{Float64}, Ptr{Float64}, Ptr{Float64}) prob.p ncol nrow mat.colptr.-Int32(1) mat.rowval.-Int32(1) mat.nzval vec_or_null(Float64, col_lb, ncol) vec_or_null(Float64, col_ub, ncol) vec_or_null(Float64, obj, ncol) vec_or_null(Float64, row_lb, nrow) vec_or_null(Float64, row_ub, nrow)
+    end
 end
 
 function readMps(prob::CbcModel, filename::String)
     check_problem(prob)
     @assert isascii(filename)
-    @cbc_ccall readMps Cint (Ptr{Cvoid}, Ptr{UInt8}) prob.p filename
+    GC.@preserve prob filename begin
+        @cbc_ccall readMps Cint (Ptr{Cvoid}, Ptr{UInt8}) prob.p filename
+    end
 end
 
 function writeMps(prob::CbcModel, filename::String)
     check_problem(prob)
     @assert isascii(filename)
-    @cbc_ccall writeMps Cint (Ptr{Cvoid}, Ptr{UInt8}) prob.p filename
+    GC.@preserve prob filename begin
```

Comment:
So this one (and the one you commented on below) I'm not sure of. I think it could currently still be an issue if `writeMps` happens to get inlined, and given Stefan's comment, there don't seem to be future guarantees regarding input arguments anyway. Again, I'm not sure of the cost of `GC.@preserve`, but I'd assume it'd be negligible compared to the `ccall`.

Reply:
The documentation makes a strong statement about ccall arguments: https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/#Garbage-Collection-Safety-1
> ccall automatically arranges that all of its arguments will be preserved from garbage collection until the call returns. 

I interpret that to mean that if this behavior changes it can only do so In a new major Julia release. I agree that the extra preserve is unlikely to have a performance impact. I'm nit picking so that we apply consistent rules and don't confuse someone who looks at this code and wonders why we preserved this particular argument when it's not needed.

Reply:
Would it be cleaner to define a `convert(::Type{Ptr{Cvoid}},::CbcModel)` method and just pass `prob` to `ccall` instead? See https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/#Auto-conversion:-1

Reply:
I'm fully onboard with the nitpicking here since this has ramifications for many other packages. I think @andreasnoack's idea is a good one.

Reply:
Actually, according to https://docs.julialang.org/en/v1/base/c/#Base.cconvert,

> Neither `convert` nor `cconvert` should take a Julia object and turn it into a `Ptr`

So I think this should be a `Base.unsafe_convert` method.